src/pykeen/evaluation/ogb_evaluator.py

```diff
@@ -183,3 +211,48 @@ def iter_preds() -> Iterable[Tuple[ExtendedTarget, torch.Tensor, torch.Tensor]]:
             value = value.mean().item()
             result[key, ext_target, rank_type] = value
     return RankBasedMetricResults(data=result)
+
+
+def _hasher(kwargs: Mapping[str, Any]) -> int:
+    return hash((id(kwargs["model"]), kwargs["mapped_triples"].shape[0], kwargs["negatives"].shape, kwargs["target"]))
+
+
+@maximize_memory_utilization(parameter_name="slice_size", hasher=_hasher)
+@maximize_memory_utilization(hasher=_hasher)
```

Comment:
My *intention* was to replicate the current setup

1. reduce batch size until we do not encounter any OOM
2. if this fails, start reducing the slice size

However, this implementation will not work as expected, as it will re-start the batch size search for each reduced slice size 😕

Reply:
This will be fixed in https://github.com/mberr/torch-max-mem/pull/10